glue/qt/custom_viewer.py

```diff
@@ -1091,6 +1090,61 @@ def value(self, layer=None, view=None):
         return self.ui.value()
 
 
+class TextBoxElement(FormElement):
+    state = wp.ValueProperty('ui')
+
+    def _build_ui(self):
+        w = QtGui.QLineEdit()
+        w.textChanged.connect(nonpartial(self.changed))
+        self._widget = w
+        if self.params.startswith('str('):
+            self.set_value(eval(self.params))
+        return w
+
+    def value(self, layer=None, view=None):
+        return self._widget.text()
+
+    def set_value(self, val):
+        self._widget.setText(str(val))
+
+    @classmethod
+    def recognizes(cls, params):
+        try:
+            if params.startswith('str'):
+                return True
+        except AttributeError:
+            pass
+
+
+class FloatElement(FormElement):
+    state = wp.ValueProperty('ui')
+
+    def _build_ui(self):
+        w = QtGui.QLineEdit()
+        w.textChanged.connect(nonpartial(self.changed))
+        self._widget = w
+        if self.params.startswith('float('):
+            self.set_value(eval(self.params))
+        return w
+
+    def value(self, layer=None, view=None):
+        try:
+            return float(self._widget.text())
+        except ValueError:
+            return None
+
+    def set_value(self, val):
+        self._widget.setText(str(val))
+
+
+    @classmethod
+    def recognizes(cls, params):
+        try:
+            if params.startswith('float'):
```

Comment:
We should think about how exactly float and text widgets should be recognized -- what you have might be fine, but it's an important decision. Should we be considering other alternatives to 'float()' and 'str()'?